pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -319,59 +319,25 @@ protected void internalGrantPermissionsOnTopic(final AsyncResponse asyncResponse
                 });
     }
 
-    private CompletableFuture<Void> revokePermissionsAsync(String topicUri, String role, boolean force) {
-        return namespaceResources().getPoliciesAsync(namespaceName).thenCompose(
-                policiesOptional -> {
-                    Policies policies = policiesOptional.orElseThrow(() ->
-                            new RestException(Status.NOT_FOUND, "Namespace does not exist"));
-                    if (!policies.auth_policies.getTopicAuthentication().containsKey(topicUri)
-                            || !policies.auth_policies.getTopicAuthentication().get(topicUri).containsKey(role)) {
-                        log.warn("[{}] Failed to revoke permission from role {} on topic: Not set at topic level {}",
-                                clientAppId(), role, topicUri);
-                        if (force) {
-                            return CompletableFuture.completedFuture(null);
-                        } else {
-                            return FutureUtil.failedFuture(new RestException(Status.PRECONDITION_FAILED,
-                                    "Permissions are not set at the topic level"));
-                        }
-                    }
```

Comment:
What is the justification for removing the check that the policies exist? This appears to remove the need for the `force` argument.

Reply:
yes, it' nothing to do with `force` argument.
We need to call `AuthorizationService ` to revoke all the permissions. Before #20456, we revoke the permissions by modification the namespace policies directly.